File: src/usecases/todo.factory.ts
import { TodoDto } from './todo.dto';
import { TodoDto as AdapterTodoDto } from '../interfaceAdapters/todo.dto';
import { AddTodoDto } from './addTodo.dto';
import { AddTodoDto as AdapterAddTodoDto } from '../interfaceAdapters/repositories/addTodo.dto';

export interface TodoFactory {
  convertToTodoDto(todoDto: TodoDto): AdapterTodoDto;

  convertToAddTodoDto(addTodoDto: AddTodoDto): AdapterAddTodoDto;
}

File: src/main/java/com/todoapp/usecases/factory/TodoFactory.java
package com.todoapp.usecases.factory;

import com.todoapp.usecases.dto.AddTodoDto;
import com.todoapp.usecases.dto.PartialTodoDto;
import com.todoapp.usecases.dto.TodoDto;
import com.todoapp.usecases.dto.TodoListDto;
import com.todoapp.entities.Todo;

import java.util.List;

public interface TodoFactory {
    Todo createTodo(AddTodoDto addTodoDto);

    TodoDto createTodoDto(Todo todo);

    PartialTodoDto createPartialTodoDto(Todo todo, String[] fields);

    TodoListDto createTodoListDto(List<Todo> todos);
} 